iree/turbine/kernel/wave/analysis/partition_strided_operators.py

```diff
@@ -50,9 +50,11 @@ def get_vector_shape(
 def _get_symbolic_shape_and_vector_shapes(
     custom: CustomOp,
 ):
-    register_shape = custom.register_type.symbolic_shape
     vector_shapes = custom.vector_shapes
     memory_shape = custom.memory_type.symbolic_shape
+    if isinstance(custom, Read):
```

Comment:
I've started digging into these partition strided operators pass, could you explain what the objective of this function is? (nit: maybe add a doc)
I am unclear why we want the register_shape in certain cases and in other the memory_shape.
Would this be a good place to assert that shapes should agree?
If they don't what is the expected behavior?